addon/components/bs-button.d.ts

```diff
@@ -0,0 +1,54 @@
+import Component from '@glimmer/component';
+
+export type BsButtonType = 'primary' | 'secondary' | 'success' | 'danger' | 'warning' | 'info' | 'light' | 'dark';
+
+interface BsButtonSignature {
+  Element: HTMLButtonElement;
+  Args: {
+    active?: boolean;
+    block?: boolean;
+    bubble?: boolean;
+    buttonType?: 'button' | 'submit';
+    defaultText?: string;
+    fulfilledText?: string;
+    icon?: string;
+    iconActive?: string;
+    iconInactive?: string;
+    outline?: boolean;
+    pendingText?: string;
+    preventConcurrency?: boolean;
+    rejectedText?: string;
+    reset?: boolean;
+    size?: string;
+    type?: BsButtonType;
+    value?: unknown;
+    onClick?: (e?: Event, value?: unknown) => void;
```

Comment:
Can return a `Promise` to control pending state of the button.

```suggestion
    onClick?: (e?: Event, value?: unknown) => Promise<void> | void;
```